src/test/java/net/sf/jabref/model/groups/ExplicitGroupTest.java

```diff
@@ -20,17 +21,36 @@ public void testToStringSimple() {
     @Test
     public void toStringDoesNotWriteAssignedEntries() {
         ExplicitGroup group = new ExplicitGroup("myExplicitGroup", GroupHierarchyType.INCLUDING, ',');
-        group.add(makeBibtexEntry());
+        group.add(new BibEntry());
         assertEquals("ExplicitGroup:myExplicitGroup;2;", group.toString());
     }
 
-    private BibEntry makeBibtexEntry() {
-        BibEntry e = new BibEntry(IdGenerator.next(), BibtexEntryTypes.INCOLLECTION.getName());
-        e.setField("title", "Marine finfish larviculture in Europe");
-        e.setField("bibtexkey", "shields01");
-        e.setField("year", "2001");
-        e.setField("author", "Kevin Shields");
-        return e;
+    @Test
+    public void testEntryStorageSingleGroups() {
+        BibEntry entry = new BibEntry();
+        ExplicitGroup group = new ExplicitGroup("myExplicitGroup", GroupHierarchyType.INDEPENDENT, ',');
+        group.add(entry);
+        assertEquals(Optional.of("myExplicitGroup"), entry.getField(FieldName.GROUPS));
+    }
+
+    @Test
+    public void testEntryStorageTwoGroups() {
+        BibEntry entry = new BibEntry();
+        ExplicitGroup group = new ExplicitGroup("myExplicitGroup", GroupHierarchyType.INDEPENDENT, ',');
+        ExplicitGroup group2 = new ExplicitGroup("myExplicitGroup2", GroupHierarchyType.INDEPENDENT, ',');
+        group.add(entry);
+        group2.add(entry);
+        assertEquals(Optional.of("myExplicitGroup, myExplicitGroup2"), entry.getField(FieldName.GROUPS));
+    }
+
+    @Test
+    public void testGroupStorageAlreadyInGroup() throws Exception {
```

Comment:
please remove `test` prefix from test names and add the expected outcome to the name (e.g. addToAlreadyPresentGroupDoesNotChangeEntry)